File: agent/python_dotnet/perpernet/protocol/cache/execution_data.py
from collections import OrderedDict
from pyignite import GenericObjectMeta
from pyignite.datatypes import String, BoolObject, ObjectArrayObject


class ExecutionData(
    metaclass=GenericObjectMeta,
    schema=OrderedDict(
        [
            ("agent", String),
            ("instance", String),
            ("delegate", String),
            ("finished", BoolObject),
            ("parameters", ObjectArrayObject),
            ("result", ObjectArrayObject),
            ("error", String),
        ]
    ),
):
    pass

File: fabric/src/main/kotlin/cache/ExecutionData.kt
package com.obecto.perper.fabric.cache

class ExecutionData(
    var agent: String,
    var instance: String,
    var delegate: String,
    var finished: Boolean,
    var parameters: Array<Any>? = null,
    var results: Array<Any>? = null,
    var error: String? = null
)
